programs/scope/src/lib.rs

```diff
@@ -145,7 +207,20 @@ pub struct OraclePrices {
 pub struct OracleMappings {
     pub price_info_accounts: [Pubkey; MAX_ENTRIES],
     pub price_types: [u8; MAX_ENTRIES],
-    pub _reserved2: [u64; MAX_ENTRIES],
+    pub twap_source: [u16; MAX_ENTRIES], // meaningful only if type == TWAP; the index of where we find the TWAP
+    pub use_twap: [u8; MAX_ENTRIES],     // true or false
+    pub _reserved1: [u8; MAX_ENTRIES],
+    pub _reserved2: [u32; MAX_ENTRIES],
+}
+
+impl OracleMappings {
+    pub fn should_use_twap(&self, token: usize) -> bool {
```

Comment:
```suggestion
    pub fn is_twap_enabled(&self, token: usize) -> bool {
```

Or maybe another name...This does not indicate we "should use twap" but more that we need to refresh twap on price updates.